python/ray/data/_internal/execution/operators/map_transformer.py

```diff
@@ -140,32 +141,78 @@ def create_map_transformer_from_block_fn(
     """
     return MapTransformer(
         [
-            MapTransformFn(
-                block_fn,
-                MapTransformFnDataType.Block,
-                MapTransformFnDataType.Block,
-            )
+            BlockMapTransformFn(block_fn),
         ],
         init_fn,
     )
 
 
-# Below are util `MapTransformFn`s for converting input/output data.
+# Below are subclasses of MapTransformFn.
+
+
+class RowMapTransformFn(MapTransformFn):
+    """A rows-to-rows MapTransformFn."""
+
+    def __init__(self, row_fn: MapTransformCallable[Row, Row]):
+        self._row_fn = row_fn
+        super().__init__(
+            MapTransformFnDataType.Row,
+            MapTransformFnDataType.Row,
+        )
+
+    def __call__(self, input: Iterable[Row], ctx: TaskContext) -> Iterable[Row]:
+        yield from self._row_fn(input, ctx)
+
+    def __repr__(self) -> str:
```

Comment:
where is this being used?

Reply:
Currently not being used. I only used this when debugging,  and decided to keep it as having a better repr won't hurt.

Reply:
now it's used in the unit test.